src/txn/txn_rollback_to_stable.c

```diff
@@ -124,7 +133,73 @@ __txn_abort_newer_insert(
     WT_INSERT *ins;
 
     WT_SKIP_FOREACH (ins, head)
-        __txn_abort_newer_update(session, ins->upd, rollback_timestamp);
+        if (ins->upd != NULL)
+            __txn_abort_newer_update(session, ins->upd, rollback_timestamp);
+}
+
+/*
+ * __txn_abort_row_ondisk_kv --
+ *     Fix the on-disk row K/V version according to the given timestamp.
+ */
+static int
+__txn_abort_row_ondisk_kv(
+  WT_SESSION_IMPL *session, WT_PAGE *page, WT_ROW *rip, wt_timestamp_t rollback_timestamp)
+{
+    WT_CELL_UNPACK *vpack, _vpack;
+    WT_DECL_RET;
+    WT_ITEM buf;
+    WT_PAGE_MODIFY *mod;
+    WT_UPDATE *upd, **upd_entry;
+    size_t size;
+
+    vpack = &_vpack;
+    __wt_row_leaf_value_cell(session, page, rip, NULL, vpack);
+    if (vpack->start_ts > rollback_timestamp) {
+        /* No visible updates in the history store, may be a newly inserted key. Remove it. */
```

Comment:
The comment is hard to understand. The part about "may be a newly inserted key" leads me to wonder what else this key could be. I suggest simplifying with:
There are no visible updates in the history store for this key. Remove it.

Reply:
Fixed.